libsignal-service-hyper/examples/registering.rs

```diff
@@ -0,0 +1,138 @@
+use std::str::FromStr;
+
+use libsignal_service::configuration::{ServiceCredentials, SignalServers};
+use libsignal_service::prelude::phonenumber::PhoneNumber;
+use libsignal_service::provisioning::{
+    generate_registration_id, ConfirmCodeMessage, ConfirmCodeResponse,
+    ProvisioningManager, VerificationCodeResponse,
+};
+use libsignal_service::push_service::{PushService, ServiceError};
+use libsignal_service::USER_AGENT;
+
+use libsignal_service_hyper::prelude::HyperPushService;
+
+use rand::RngCore;
+
+#[tokio::main]
+async fn main() {
+    let phonenumber = let_user_enter_phone_number();
+    let password = let_user_enter_password();
+    let use_voice = does_user_want_voice_confirmation();
+    let captcha = let_user_solve_captcha();
```

Comment:
Since it's an example, you could also opt to ask these things on CLI, then a developer can actually run the example too. If you don't feel like implementing that, that's fine by me!

Reply:
Honestly, I'll just merge this as-is. We can doodle on that later.

Reply:
Agreed. If I find time, I might get back to it and provide a PR...